doc/zmq_close.txt

```diff
@@ -21,22 +21,18 @@ behaviour for discarding messages sent by the application with _zmq_send()_ but
 not yet physically transferred to the network depends on the value of the
 _ZMQ_LINGER_ socket option for the specified 'socket'.
 
+_zmq_close()_ must be called exactly once for each socket. If it is never called,
+_zmq_ctx_term()_ will block forever. If it is called multiple times for the same 
+socket or if _socket_ does not point to a socket, the behaviour is undefined.
```

Comment:
Sockets are either thread safe, so the tag is set to deadbeef inside a critical section, or they are not and so only one zmq_close can be called at any one time on a socket, which means setting the tag will work and subsequent calls will return ENOTSOCK.

Is this not working correctly in some cases?

Reply:
See my comment above, third paragraph.